docs/mine/lotus/message-pool.md

```diff
@@ -0,0 +1,76 @@
+
+If the ticket quality is sufficiently high, then a greedy selection algorithm is used instead that simply picks dependent chains of maximum reward. Note that pending message chains from priority addresses are always selected, regardless of their profitability.
+
+## Gas, fees, limits and caps
+
+When a message is executed it consumes _gas_. The total gas by a message directly affects the cost to place that message in the blockchain, that the sender will have to pay.
+
+In the Filecoin Network, a dynamic **_BaseFee_**, measured in attoFIL/gas units, specifies how much FIL gets burned for every message based on the _gas_ they used for their execution. The _BaseFee_ gets automatically updated according to the network congestion parameters (block sizes). The current value can be obtained from one of the [block explorers](../../get-started/explore-the-network.md) or by looking at the current head:
+
+```sh
+# Will print the last BaseFee in attoFIL
+lotus chain head | xargs lotus chain getblock | jq -r .ParentBaseFee
+```
+
+Apart from the _BaseFee_, every message in the message pool has the following gas-related fields:
+
+- **_GasLimit_**: it is measured in gas units. It a hard limit on the amount of gas that a message's execution can consume. A message consumes gas for every fundamental operation it takes (see [prices](https://github.com/filecoin-project/lotus/blob/d678fe4bfa5b4c70bcebd46cdc38aafc452b42d1/chain/vm/gas.go#L87)). Messages can fail if they runs out of gas fails, at which point every modification to the state is reverted. Whether the execution is successful or not, the miner is given a reward calculated as _GasLimit _ GasPremium\*.
+- **_GasPremium_**: it is measured in attoFIL/gas units. It indicates how much a miner earns as their reward for including a message. A message typically earns its miner _GasLimit \* GasPremium_ attoFIL.
+- **_GasFeeCap_**: it is measured attoFIL/gas units. Its purpose is to establish a hard limit on the amount of FIL a message can cost its sender. A sender is guaranteed that a message will never cost them more than _GasLimit \* GasFeeCap_, not including any value that the message includes for its recipient. The total fee (in attoFIL/gas) for a message is _GasPremium + BaseFee_. Given that _GasPremium_ is set by the sender, _GasFeeCap_ essentially serves as a safeguard against a high, unexpected _BaseFee_.
+
+Note that if a message's _GasFeeCap_ is lower than the _BaseFee_, then the remainder comes from the miner (as a penalty). If _BaseFee + GasPremium_ is less than the message's _GasFeeCap_, then the miner might not earn the entire _GasLimit \* GasPremium_ reward.
```

Comment:
@arajasek I hope I have been able to correctly port our conversation here.

Reply:
I'm coming from the slack thread where this sentence confused me.  After checking out the [code](https://github.com/filecoin-project/lotus/blob/efc1b24f4ea234e95fdc038d83e17400671c0ade/chain/vm/burn.go#L70) I think the "less than" should be "greater than". I think a clearer way to say this is "If GasFeeCap + GasPreium is greater than the message's GasFeeCap the miner's reward becomes GasLimit * (GasFeeCap - BaseFee)"

(The above isn't exactly true becasue GasFeeCap - BaseFee can go negative but it might be more confusing to call this out here.  I think the right overall expression is `GasLimit * min(GasPremium, (max(0, GasFeeCap - BaseFee))`

Reply:
`"If GasFeeCap + GasPreium is greater than the message's GasFeeCap the miner's reward becomes GasLimit * (GasFeeCap - BaseFee)"`

You mean "If _BaseFee + GasPremium_" I guess :P. Taking your suggestion.